generation/framework/utils.py

```diff
@@ -165,3 +165,21 @@ def run_command(executable, env, *args, **kwargs):
         if error:
             logger.info('Error:\n{}'.format(error))
     return code, out, error
+
+
+def check_outdir(outputdict):
+    """ensure that the output directory exists"""
+    for output in outputdict.values():
+        try:
+            output.parent.touch()
+        except IOError:
+            logger.error("Output target {} doesn't exist!".format(output.parent))
+            output.makedirs()
```

Comment:
What will happen if you call makedir on an already existing path? I think changing the try to makeidr would make more sense.

Reply:
Could be tested, but it works currently and there is no need for the effort. I kept it basically as it was. Maybe something for the future.